article/app/controllers/LiveBlogController.scala

```diff
@@ -14,75 +14,127 @@ import play.api.mvc._
+    mapModel(path, Canonical)((page, blocks) => render(path, page, blocks))
+  }
+
+  def renderEmail(path: String): Action[AnyContent] = {
     Action.async { implicit request =>
-      mapModel(path, ArticleBlocks) {
-        render(path, _)
-      }
+      mapModel(path, ArticleBlocks)((page, blocks) => render(path, page, blocks))
     }
+  }
 
-  def renderArticle(path: String, page: Option[String] = None, format: Option[String] = None): Action[AnyContent] =
-      Action.async { implicit request =>
-        def renderWithRange(range: BlockRange) =
-          mapModel(path, range) {
-            render(path, _)
-          }
-        page.map(ParseBlockId.fromPageParam) match {
-          case Some(ParsedBlockId(id)) => renderWithRange(PageWithBlock(id)) // we know the id of a block
-          case Some(InvalidFormat) => Future.successful(Cached(10)(WithoutRevalidationResult(NotFound))) // page param there but couldn't extract a block id
-          case None => renderWithRange(Canonical) // no page param
-        }
+  // Main entry points
+
+  def renderArticle(path: String, page: Option[String] = None, format: Option[String] = None): Action[AnyContent] = {
+    Action.async { implicit request =>
+      def renderWithRange(range: BlockRange): Future[Result] = {
+        mapModel(path, range)((page, blocks) => render(path, page, blocks))
       }
 
-  def renderJson(path: String, lastUpdate: Option[String], rendered: Option[Boolean], isLivePage: Option[Boolean]): Action[AnyContent] = {
+      page.map(ParseBlockId.fromPageParam) match {
+        case Some(ParsedBlockId(id)) => renderWithRange(PageWithBlock(id)) // we know the id of a block
+        case Some(InvalidFormat) => Future.successful(Cached(10)(WithoutRevalidationResult(NotFound))) // page param there but couldn't extract a block id
+        case None => renderWithRange(Canonical) // no page param
+      }
+    }
+  }
+
+  def renderJson(
+    path: String,
+    lastUpdate: Option[String],
+    rendered: Option[Boolean],
+    isLivePage: Option[Boolean]
+  ): Action[AnyContent] = {
+
     Action.async { implicit request =>
       val range = getRange(lastUpdate, rendered)
-
       mapModel(path, range) {
-        case liveblog: LiveBlogPage if rendered.contains(false) => getJsonForFronts(liveblog)
-        case liveblog: LiveBlogPage => getJson(path, liveblog, range, isLivePage)
-        case minute: MinutePage => render(path, minute)
+        case (liveblog: LiveBlogPage, blocks) => getJson(path, liveblog, range, isLivePage, blocks)
+        case (minute: MinutePage, blocks) => render(path, minute, blocks)
         case _ => Future { Cached(600)(WithoutRevalidationResult(NotFound)) }
       }
+    }
+  }
+
+  // Helper methods
+
+  private[this] def render(
+    path: String,
+    page: PageWithStoryPackage,
+    blocks: Blocks
+  )(implicit request: RequestHeader): Future[Result] = {
+    val isMinute = page.isInstanceOf[MinutePage]
 
+    Future.successful {
```

Comment:
I couldn't see why this needed to be `Future` before (as that pushes stuff onto a separate EC) but @MatthewJWalls do you know if there was a reason for this?

Reply:
I think it's fine to be Future. successful. I seem to remember having some trouble getting scala to be happy with the types when we originally split up the article controller and liveblog controller, but there wasn't any concurrency/performance/etc reasons for using a straight Future I think.